gef.py

```diff
@@ -8107,9 +8119,20 @@ def do_invoke(self, _: List[str], **kwargs: Any) -> None:
 
         addr = align_address(parse_address(args.location))
         size, fcode = self.SUPPORTED_SIZES[self.format]
+        values = args.values
+
+        # only support env var as values in patch byte command
+        if size == 1: 
+            if "$" in values[0]:
+                envar = values[0]
+                try:
+                    values = str(gdb.parse_and_eval(envar)).replace("\"", "").split(" ")
```

Comment:
I think this is still too complicated for what you try to achieve: 

A simpler (more pythonic) way would be to simply store the raw data in the convenience variable. No need for parsing or even bitlen, just make a convenience variable from `gef.read.memory` It would go like:

```
gef➤  pf --lang hex -l 16 $rsp
01000000000000003fe4ffffff7f000000000000000000004be4ffffff7f0000
gef➤  pf --lang raw -l 16 $rsp
Saved as $_gef0
gef➤  display/x $_gef0
3: /x $_gef0 = {0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3f, 0xe4, 0xff, 0xff, 0xff, 0x7f, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x4b, 0xe4, 0xff, 0xff, 0xff, 0x7f, 0x0, 0x0, 0x0}
```

Which means in turn, your `PatchCommand byte` would be much simpler, by simply checking that the data parameter is an convenience variable, if so just do a `gef.memory.write`.

```
gef➤  patch byte $rip $_gef0
```

This approach has also a huge advantage that it will be seen by GDB as a native array, allowing modification:

```
gef➤  display/x $_gef0[0]
4: /x $_gef0[0] = 0x1
gef➤  set $_gef0[0]=0x42
gef➤  display/x $_gef0[0]
5: /x $_gef0[0] = 0x42
```

Note that currently, the "storing" part can already be achieved natively:

```
gef➤  pi gef_convenience( gef.memory.read(0x007fffffffe150, 16) )
'$_gef1'
gef➤  display/x $_gef1
6: /x $_gef1 = {0x62, 0x27, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x3f, 0xe4, 0xff, 0xff, 0xff, 0x7f, 0x0, 0x0, 0x27, 0x0}
```

Reply:
done!